packages/core/src/components/SingleSelect/FlatVariant/FlatVariant.styled.tsx

```diff
@@ -0,0 +1,132 @@
+import { centerAligned, InjectClassName, styled } from '@medly-components/utils';
+import { css } from '@medly-components/utils';
+import { rgba } from 'polished';
+import Label from '../../Label';
+import { FlatVariantProps } from './types';
+
+const getButtonStyle = (
+    valueColor: string,
+    labelColor: string,
+    errorColor: string,
+    activatedColor: string,
+    bgColor: string,
+    areOptionsVisible: boolean,
+    errorText: string
+) => css`
+    color: ${errorText ? errorColor : valueColor};
+    background: ${areOptionsVisible ? activatedColor : bgColor};
+    * {
+        fill: ${errorText ? errorColor : valueColor};
+    }
+    ${Label.Style} {
+        color: ${labelColor};
+    }
+`;
+
+const buttonStyle = ({ theme, areOptionsVisible, errorText }: FlatVariantProps) => {
+    const { bgColor, labelColor, valueColor } = theme.singleSelect.variant.flat;
+    return css`
+        &:disabled {
+            ${getButtonStyle(
+                valueColor.disabled,
+                labelColor.disabled,
+                valueColor.error,
+                bgColor.activated,
+                bgColor.disabled,
+                areOptionsVisible,
+                errorText
+            )}
+        }
+
+        &:not(:disabled) {
+            &:not(:hover) {
+                ${getButtonStyle(
+                    valueColor.default,
+                    labelColor.default,
+                    valueColor.error,
+                    bgColor.activated,
+                    bgColor.default,
+                    areOptionsVisible,
+                    errorText
+                )}
+            }
+            &:active {
+                ${getButtonStyle(
+                    valueColor.pressed,
```

Comment:
All the props can be read in `getButtonStyle` function itself. You can just pass one more prop with it as state: "default" | "hovered" | "pressed" | "selected".